symfony/plugins/orangehrmAttendancePlugin/lib/dao/AttendanceDao.php

```diff
@@ -516,6 +516,26 @@ public function searchAttendanceRecords($employeeIds = null, $employeementStatus
       
     }
 
+    /**
+     * @param int $employeeId
+     * @param string $state
+     * @return array|bool|Doctrine_Record|float|int|mixed|string|null
+     * @throws DaoException
+     */
+    public function getLatestPunchInRecord(int $employeeId, string $state)
```

Comment:
Fetching the recently added record to the database and considering it as the latest punch in record is correct in an ideal scenario.

But there can be cases that the recently added record is not the latest record.

Therefore the "orderBy" logic should be changed to order by the punch_in_utc_time instead of the id.